tests/end-to-end/external-solution-integration/pkg/testsuite/deploy_function.yaml.go

```diff
@@ -0,0 +1,187 @@
+    });	
+}
+function getGateway() {
+	if (legacy) {
+		return process.env.GATEWAY_URL;
+	} else {
+		let envKey = Object.keys(process.env).find(val => val.endsWith('_GATEWAY_URL'));
+		return process.env[envKey]
+	}
+}
+
+module.exports = { main: function (event, context) {	
+	console.log("==============================");	
+	console.log("Legacy:           ", legacy)
+	console.log("Expected Payload: ", expectedPayload)
+	console.log("==============================");
+	console.log("Received event: ");	
+	console.log(JSON.stringify(event, null, 2));	
+	console.log("==============================");	
+	console.log("Received context:");	
+	console.log(JSON.stringify(context, null, 2));	
+	console.log("==============================");	
+	
+	const gateway = getGateway()
+    if (gateway === undefined) {
+		throw new Error("gateway is undefined")	
+	}
+    if (event["data"] !== expectedPayload) {	
+		throw new Error("Payload not as expected")	
+    }
+	return new Promise((resolve, reject) => {	
+		const url = gateway + "/counter";	
+		console.log("Counter URL: ", url);	
+		sendReq(url, resolve, reject);	
+	}).then(resolved, rejected);	
+} };	
+`
+const functionDeps = `{"dependencies":{"request": "^2.88.0", "circular-json": "^0.5.9"}}`
+
+// DeployFunction deploys function to the cluster. The function will do PUT /counter to connected application upon receiving
+// an event
+type DeployFunction struct {
+	client          dynamic.ResourceInterface
+	name            string
+	port            int
+	expectedPayload string
+	legacy          bool
+}
+
+var _ step.Step = &DeployFunction{}
+
+func NewDeployFunction(name, expectedPayload string, port int, client dynamic.ResourceInterface, legacy bool) *DeployFunction {
+	return &DeployFunction{
+		client:          client,
+		name:            name,
+		port:            port,
+		legacy:          legacy,
+		expectedPayload: expectedPayload,
+	}
+}
+
+func (s *DeployFunction) Name() string {
+	return fmt.Sprintf("Deploy function %s", s.name)
+}
+
+func (s *DeployFunction) Run() error {
+	if err := s.createFunction(); err != nil {
+		return err
+	}
+
+	if err := retry.Do(s.isFunctionReady); err != nil {
+		return errors.Wrap(err, "function not ready")
+	}
+
+	return nil
+}
+
+func (s *DeployFunction) createFunction() error {
+	function := &serverless.Function{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Function",
+			APIVersion: serverless.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: s.name,
+		},
+		Spec: serverless.FunctionSpec{
+			Source: fmt.Sprintf(functionFmt, s.expectedPayload, s.legacy),
+			Deps:   functionDeps,
+		},
+	}
+
+	unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(function)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.client.Create(&unstructured.Unstructured{Object: unstructuredObj}, metav1.CreateOptions{})
+	return err
+}
+
+func (s *DeployFunction) Cleanup() error {
+	return s.client.Delete(s.name, &metav1.DeleteOptions{})
```

Comment:
Previously there was functionality to wait until the resource is deleted ... can we have it back?

this is how it is done elsewhere:
```
return helpers.AwaitResourceDeleted(func() (interface{}, error) {
		return s.serviceInstances.Get(s.name, metav1.GetOptions{})
	})
```

Reply:
Good point, thanks